sssd_test_framework/roles/ipa.py

```diff
@@ -1597,3 +1610,103 @@ def __get_info(self, info: str | NFSExport | IPAAutomountMap | None) -> str | No
             return info.name
 
         return info
+
+
+class IPAPasswordPolicy(IPAObject):
+    """
+    Password policy management.
+    """
+
+    def __init__(self, role: IPA, name: str):
+        """
+        :param role: IPA host object.
+        :type role: IPAHost
+        :param name: Name of target object.
+        :type name: str
+        """
+        super().__init__(role, name, command_group="pwpolicy")
+
+    def complexity(self, enable: bool) -> IPAPasswordPolicy:
+        """
+        Enable or disable password complexity.
+
+        :param enable: Enable or disable password complexity.
+        :type enable: bool
+        :return: IPAPasswordPolicy object.
+        :rtype: IPAPasswordPolicy
+        """
+        if enable:
+            attrs: CLIBuilderArgs = {
+                "dictcheck": (self.cli.option.VALUE, "True"),
+                "usercheck": (self.cli.option.VALUE, "True"),
+                "minlength": (self.cli.option.VALUE, 8),
+                "minclasses": (self.cli.option.VALUE, 5),
+                "priority": (self.cli.option.VALUE, 1),
+            }
+            self._add(attrs)
```

Comment:
Why are you calling `self._add()` here but `self._modify()` in the `else` section?

Reply:
It's the way the command is constructed. `IPARole._add` is `ipa pwpolicy add` , while `IPARole._modify` is `ipa pwpolicy modify`. If enabled, we are adding the values that don't exist, to disable it, we are modifying the existing values.

Reply:
Nice catch, I'll go ahead and add something to check to see if the attribute exists and decide which function to use.